app/lib/features/chat/providers/notifiers/media_chat_notifier.dart

```diff
@@ -93,4 +99,36 @@ class MediaChatNotifier extends StateNotifier<MediaChatState> {
       }
     }
   }
+
+  static Future<File?> getThumbnailData(String mediaPath) async {
+    try {
+      final tempDir = await getTemporaryDirectory();
+      final videoName = mediaPath.split('/').last.split('.').first;
+      final destPath = p.join(tempDir.path, '$videoName.jpg');
+      final destFile = File(destPath);
+
+      if (await destFile.exists()) {
+        return destFile;
+      }
+
+      final thumbnailGenerated =
+          await FcNativeVideoThumbnail().getVideoThumbnail(
+        srcFile: mediaPath,
+        destFile: destPath,
+        width: 300,
+        height: 300,
+        keepAspectRatio: true,
+        format: 'jpeg',
+        quality: 90,
+      );
+
+      if (thumbnailGenerated) {
+        return destFile;
+      }
```

Comment:
ehm ... while I appreciate the attempt to just do things on the flutter side, taking any data, transforming it and then putting it on the filesystem is not an acceptable solution for us as it means we are potentially storing end-to-end-encrypted information in plain on the device (known as unencrypted at rest). This is highly insecure and puts our users at risk. As layed out in https://github.com/acterglobal/a3/issues/1586 as well.

I am _okay_ with temporarily accepting this PR if a FIXME linking to the above issue is added and a comment is left on that issue pointing to this additional code base to be fixed as well.

Reply:
Yes that's true and agree with your thoughts.

Once we implement thumb management from rust sdk (https://github.com/acterglobal/a3/issues/1250), this code will change this is temporarily solution to avoid unwanted performance and sizing issue on chat messages. I will required FIXME not here so that we can take care of it in future.

Probably `getTemporaryDirectory` will not be directly accessible to end users from file explore so want be that much security issue but ya surely in future we need to avoid this.